src/silicone/database_crunchers/rms_closest.py

```diff
@@ -247,7 +300,13 @@ def _filter_for_overlap(df1, df2, cols):
     """
     lead_data = df1.data.set_index(cols)
     follow_data = df2.data.set_index(cols)
-    shared_indices = [ind for ind in lead_data.index if ind in follow_data.index]
+    shared_indices = [
+        ind
+        for ind in follow_data.index
+        if lead_data.index.tolist().count(ind) == len(leaders)
```

Comment:
Is there a way to write this which makes it more obvious what it's doing?

Reply:
I can't see anything tremendously simplifying but I can certainly add more comments and documentation.